packages/dart_frog_auth/lib/src/dart_frog_auth.dart

```diff
@@ -35,10 +41,20 @@ extension on Map<String, String> {
 ///
 /// If the given function returns null for the given username and password,
 /// the middleware will return a `401 Unauthorized` response.
+///
+/// By default, this middleware will apply to all routes. You can change this
+/// behavior by providing a function that returns a boolean value based on the
+/// [RequestContext]. If the function returns false, the middleware will not
+/// apply to the route and the call will have authentication validation.
 Middleware basicAuthentication<T extends Object>({
   required Future<T?> Function(String, String) userFromCredentials,
+  ApplyToRoute applyToRoute = _defaultApplyToRoute,
```

Comment:
```suggestion
  ApplyToRoute applyTo = _defaultApplyToRequest,
```

or 

```suggestion
  ApplyToRoute appliesTo = _defaultAppliesToRequests,
```

or

```suggestion
  ApplyToRoute appliesToRequests = _defaultAppliesToRequests,
```


I personally prefer the plural ones since they infer it can be applied to more than one type of request.

Reply:
Related to https://github.com/VeryGoodOpenSource/dart_frog/pull/715#discussion_r1239582940.